app/dao/role.dao.js

```diff
@@ -0,0 +1,158 @@
+var request = require('request');
+var config = require('../config/config');
+var q = require('q');
+
+var url = config.api_url_dev + 'role';
+
+function parseAll(body) {
+    var deferred = q.defer();
+    var roles = JSON.parse(body);
+    if (roles) {
+        deferred.resolve(roles);
+    }
+    deferred.resolve([]);
+    return deferred.promise;
+};
+
+function parseOne(body) {
+    var deferred = q.defer();
+    parseAll(body)
+        .then(function(roles) {
+            if (roles.length) {
+                deferred.resolve(roles);
+            }
+            deferred.resolve(null);
+        });
+};
+
+// res[0]: the response
+// res[1]: the body
+// res[2]: the error
+function parseResponse(response) {
+    return q.promise(function(resolve, reject) {
+        if (!response) {
+            reject(new Error('Invalid response format'));
+        }
+        if (response[2]) {
+            reject(response[2]);
+        }
+        resolve(response);
+    });
+};
+
+function parseDelete(response) {
+    return q.promise(function(resolve, reject) {
+        switch (response[0].statusCode) {
```

Comment:
To make this a little more terse you can put `return` infront of all calls to `resolve` and `reject` and skip all the `break`s